volumina/volumeEditorWidget.py

```diff
@@ -245,6 +245,12 @@ def onShapeChanged():
                 cropMin = (self.editor.posModel.time,self.editor.cropModel._crop_extents[0][0],self.editor.cropModel._crop_extents[1][0],self.editor.cropModel._crop_extents[2][0],0)
                 self.quadViewStatusBar.updateShape5Dcropped(cropMin,self.editor.posModel.shape5D)
 
+            timeDim = self.editor.posModel.shape5D[0]
```

Comment:
New code deals with changes of data with T and data without T.
It is in the same block of code as:

"            singletonDims = filter( lambda (i,dim): dim == 1, enumerate(self.editor.posModel.shape5D[1:4]) )
..."
that deals with changes of XY vs. XYZ data.

Having in mind that _setupVolumeExtent() is only called from this line we could merge the whole proc, or leave it as it is with a minimal change of code base.